lib/engine/health.ex

```diff
@@ -14,18 +14,27 @@ defmodule Engine.Health do
   @hackney_pools [:default, :arinc_pool]
   @default_period_ms 60_000
   @failed_request_limit 5
+  @process_health_interval_ms 5_000
+  @process_metrics ~w(memory binary_memory heap_size total_heap_size message_queue_len reductions)a
 
   def start_link(opts \\ []) do
     GenServer.start_link(__MODULE__, opts)
   end
 
+  @spec init(keyword) :: {:ok, Engine.Health.t()}
   def init(opts) do
     period_ms = Keyword.get(opts, :period_ms, @default_period_ms)
     network_check_mod = Keyword.get(opts, :network_check_mod, Engine.NetworkCheck.Hackney)
     restart_fn = Keyword.get(opts, :restart_fn, Application.get_env(:realtime_signs, :restart_fn))
 
     {:ok, timer_ref} = :timer.send_interval(period_ms, self(), :health_check)
 
+    Process.send_after(
```

Comment:
Since this blindly repeats on an interval, you can do `{:ok, _} = :timer.send_interval(@process_health_interval_ms, self(), :process_health)` like the line above, and then the `handle_info` for `:process_health` won't need to worry about sending itself a message later.

I'm not super sure why we're storing the `timer_ref` of the other interval. It doesn't look like we do anything with it, so you probably don't need to store the process health one.

Reply:
My usage here was influenced by this part of the Erlang efficiency guide: https://www.erlang.org/doc/efficiency_guide/commoncaveats.html#timer-module. Maybe in this case we're not so worried about creating too many timers, or overloading this process with logging requests that it can't handle, since they're being sent infrequently?

Reply:
Yeah, I think in this case we can go with the simple `:timer.send_interval/3` approach.